### PR TITLE
Refactor finding pybind11

### PR DESCRIPTION
**Pull request by chapulina**
* Builds on top of #364 

## Summary

* Move `find_package` call to top-level `CMakeLists` together with other dependencies
* Print status messages if found / not found for debugging

The change in `src/python_pybind11/CMakeLists.txt` is mainly whitespace.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -67,15 +67,26 @@ if (SWIG_FOUND)
...
 
-  ########################################
-  # Include python
-  find_package(PythonInterp 3 REQUIRED) # change to Python3 when Bionic is EOL
-  find_package(PythonLibs QUIET)
-  if (NOT PYTHONLIBS_FOUND)
-    message (STATUS "Searching for Python - not found.")
+########################################
+# Python bindings
+find_package(PythonInterp 3 REQUIRED) # change to Python3 when Bionic is EOL
```
Comment (scpeters) [inline CMakeLists.txt:74]: have you considered `include(IgnPython)`?

* https://github.com/ignitionrobotics/sdformat/blob/sdf12/CMakeLists.txt#L62
* https://github.com/ignitionrobotics/ign-cmake/blob/ign-cmake2/cmake/IgnPython.cmake

Comment (chapulina) [reply CMakeLists.txt:74]: I have not, I was just moving existing code. Let me give that a try

Comment (chapulina) [reply CMakeLists.txt]: ca39f94

Comment (scpeters) [reply CMakeLists.txt]: it needs to be `include(IgnPython)`, not `find_package(IgnPython)`

Comment (scpeters) [reply CMakeLists.txt]: https://github.com/ignitionrobotics/ign-math/pull/360/commits/20104aed9ed38eb1cfb8b3a8c7aea6d32364b149

Comment (chapulina) [reply CMakeLists.txt]: Thanks. It's still unhappy though

```
CMake Error at /usr/lib/cmake/pybind11/FindPythonLibsNew.cmake:96 (message):
  Python config failure:

  Traceback (most recent call last):

    File "<string>", line 1, in <module>

  ImportError: cannot import name 'sysconfig' from 'distutils'
  (/usr/lib/python3.8/distutils/__init__.py)

Call Stack (most recent call first):
  /usr/lib/cmake/pybind11/pybind11Tools.cmake:16 (find_package)
  /usr/lib/cmake/pybind11/pybind11Config.cmake:100 (include)
  CMakeLists.txt:83 (find_package)
```

**Comment by j-rivero** [comment]
I think we have a missing dep in Jenkins, I'm looking into it.

**Comment by chapulina** [comment]
The `ign-math6` branch is failing too since we moved to Focal on Jenkins:

https://build.osrfoundation.org/job/ignition_math-ci-ign-math6-focal-amd64/1/console

File: CMakeLists.txt (hunk 1 above)
Comment (chapulina) [reply CMakeLists.txt]: It's green now with the dependency that was missing, thanks to @j-rivero

**Comment by chapulina** [comment]
> The ign-math6 branch is failing too since we moved to Focal on Jenkins:

This PR should fix it

**Comment by chapulina** [comment]
I broke off the Focal fix into https://github.com/ignitionrobotics/ign-math/pull/364 for easier review.

**Review by ahcorde — APPROVED** [review]

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-03-01-citadel-edifice-fortress/1313/1

**Pull request merged**